src/pages/home/sidebar/SidebarView.js

```diff
@@ -14,39 +17,64 @@ const propTypes = {
     // Safe area insets required for mobile devices margins
     insets: SafeAreaInsetPropTypes.isRequired,
 
-    // when the chat switcher is selected
+    // Current state of the chat switcher (active of inactive)
     isChatSwitcherActive: PropTypes.bool,
 
-    // Current state (active or not active) of the FAB
-    isFloatingActionButtonActive: PropTypes.bool.isRequired,
+    // Current state of the CreateMenu component (active or inactive)
+    isCreateMenuActive: PropTypes.bool,
 
-    // Callback to fire on request to toggle the FAB
-    onFloatingActionButtonPress: PropTypes.func.isRequired,
+    // Callback to fire on request to toggle the CreateMenu
+    toggleCreateMenu: PropTypes.func.isRequired,
+
+    // The data array containing the icon, text and callback information of each item
+    menuItemData: PropTypes.arrayOf(PropTypes.shape({
+        // The icon component of the item
+        icon: PropTypes.func.isRequired,
+
+        // The text content of the item
+        text: PropTypes.string.isRequired,
+
+        // Callback to fire on item press
+        onPress: PropTypes.func.isRequired,
+    })).isRequired,
 };
 
 const defaultProps = {
     isChatSwitcherActive: false,
+    isCreateMenuActive: false,
 };
 
 const SidebarView = props => (
-    <View style={[styles.flex1, styles.sidebar]}>
-        <SidebarLinks
-            onLinkClick={props.onLinkClick}
-            insets={props.insets}
-            isChatSwitcherActive={props.isChatSwitcherActive}
-        />
-        {!props.isChatSwitcherActive && (
+    <>
+        <View style={[styles.flex1, styles.sidebar]}>
+            <SidebarLinks
+                onLinkClick={props.onLinkClick}
+                insets={props.insets}
+            />
+            {!props.isChatSwitcherActive && (
             <SidebarBottom insets={props.insets} />
-        )}
-        <FAB
-            isActive={props.isFloatingActionButtonActive}
-            onPress={props.onFloatingActionButtonPress}
-            isHidden
+            )}
+            <FAB
+                isActive={props.isCreateMenuActive}
+                onPress={props.toggleCreateMenu}
+            />
+        </View>
+        <CreateMenu
+            onClose={props.toggleCreateMenu}
+            isVisible={props.isCreateMenuActive}
+            menuItemData={props.menuItemData}
```

Comment:
This is maybe deviating a bit from the original proposal, but thinking about this some more and we are drilling these props down a bit unnecessarily. Here's an alternative idea...

1. Declare the menu items inside `CreateMenu`
2. Add an `onItemSelected` callback
3. Pass a function down to `CreateMenu` instead of having to declare the items in `HomePage`.

Feel free to push back on that as we can always improve this later.

Reply:
I have to say I wasn't liking the `menuItemData` array on the `HomePage` as well.
I guess it can make it harder to get each item's callback function when they have one, but it's definitely worth trying out to see.

I put a more declarative name for the function on the `HomePage`: `onCreateMenuItemSelected`, left it as `onItemSelected` inside the `CreateMenu` component.

I wrapped `CreateMenu` with `React.memo()` now, since I don't want the `menuItemData` array to be reconstructed as often.

Reply:
Nice it looks good!